cpp-package/example/alexnet.cpp

```diff
@@ -215,7 +215,7 @@ int main(int argc, char const *argv[]) {
   args_map["label"] = NDArray(Shape(batch_size), ctx);
 
   /*with data and label, executor can be generated automatically*/
-  auto *exec = Net.SimpleBind(ctx, args_map);
+  auto exec = Net.SimpleBind(ctx, args_map);
```

Comment:
Why not use unique_ptr<Executor> rather than a stack variable?

Reply:
Ok, I see it is returning unique_ptr. ANy chance you can not use auto in this case because it's not obvious what the type is from the code line.

Reply:
I can change it for sure to the proper type. Are we going to merge this PR though? I really dislike returning unmaged pointers and would like to see this merged, even if it breaks the interface very slightly.

Reply:
It doesn't necessarily have to return a unique_ptr as long as you immediately assign it to one.  Might be a benefit for it not being returned as unique_ptr from SImpleBind() since maybe they want it to be a shared_ptr in their calling context.

Reply:
Are these pointers returns to python as object references? I think they might be...

Reply:
Oh wait, this is for cpp-package only?  I understand the "changing the interface" thing now.  Maybe bring it up on dev if you still want to change it?  But in that case, I prefer shared_ptr since this seems like something they're goingto want to save (just my personal preference).

Reply:
Thanks. I will bring it up on the list. About shared_ptr: since you can create it yourself by moving from the unique_ptr if you need a shared_ptr, shouldn't be a big issue. Actually faster to not create a shared_ptr if not needed, as it doesn't need the atomic refcount machinery.

Reply:
You may want to point out on the thread that this is an API change for call X (might as well make it for both Bind calls) etc